### DISCUSSION TITLE
How to retrieve a Stripe payment from a submission in Freeform 5.x ?

### DISCUSSION THREAD

**Discussion by croxton** [solspace/craft-freeform#1290] [Questions & Answers] [answered]
In v4 it used to be possible to get payment data from a submission ID like this in a template:

```
 {% set payments = craft.freeform.payments(submissionId) %}
```

In v5 the [documentation](https://docs.solspace.com/craft/freeform/v5/integrations/payments/#submission-data-example) suggests you should be able to do something similar:

```
 {% set payments = freeform.payments(submissionId) %}
```

However this throws an error:
```
Neither the property "payments" nor one of the methods "payments()", "getpayments()"/"ispayments()"/"haspayments()" or "__call()" exist and have public access in class "Solspace\Freeform\Variables\FreeformVariable".
```

Is this a bug or have the docs not been updated properly, and if so how do I do get hold of the payment info? Many thanks.

**Comment by croxton** [comment]
Update: I'm able to get the payment record like this:

```
{% set paymentRecord = create('Solspace\\Freeform\\Records\\Pro\\Payments\\PaymentRecord') %}
{% set payments = paymentRecord.find({'submissionId' : submissionId}).one() %}
```

However, I feel like it should be easier?

**Reply by gustavs-gutmanis** [reply]
It seems like during the massive overhaul of Freeform 5, this feature has not been ported over yet, we will add support for it again, soon.

Thanks for letting us know!

**Reply by gustavs-gutmanis** [reply]
This has been fixed in the upcoming version of Freeform, which will be released sometime soon.

**Comment by kjmartens** [comment, marked as answer]
This should now be resolved in Freeform 5.2.2+. 🙂
